tests/unit/test_launchers.py

```diff
@@ -105,10 +106,22 @@ def test_launcher_deleted_remove_tempdir(self):
 
 
 class TestFennecLauncher(unittest.TestCase):
+
+    def setUp(self):
+        self.test_root = '/sdcard/tmp'
+        self.profile = Profile()
+        self.profile_name = os.path.basename(self.profile.profile)
+        # No join because there is a path check later on
+        self.remote_profile_path = self.test_root + '/' + self.profile_name
+
+    def tearDown(self):
+        # explicit cleanup to force profile directory deletion
+        self.profile.cleanup()
```

Comment:
I would prefer to see this in the setUp method, registered with addCleanup: https://docs.python.org/2/library/unittest.html#unittest.TestCase.addCleanup
This is less verbose and will be called even if the test fail.